xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/AudioDSP.h

```diff
@@ -0,0 +1,1288 @@
+    }
+    //--------------------------------------------------------------------------
+
+  private:
+    void SetAddonStruct(KODI_HANDLE instance)
+    {
+      if (instance == nullptr)
+        throw std::logic_error("kodi::addon::CInstanceVisualization: Null pointer instance passed.");
+
+      m_instanceData = static_cast<AddonInstance_AudioDSP*>(instance);
+
+      m_instanceData->toAddon.get_capabilities = ADDON_GetCapabilities;
+      m_instanceData->toAddon.get_dsp_name = ADDON_GetDSPName;
+      m_instanceData->toAddon.get_dsp_version = ADDON_GetDSPVersion;
+      m_instanceData->toAddon.menu_hook = ADDON_MenuHook;
+
+      m_instanceData->toAddon.stream_create = ADDON_StreamCreate;
+      m_instanceData->toAddon.stream_destroy = ADDON_StreamDestroy;
+      m_instanceData->toAddon.stream_is_mode_supported = ADDON_StreamIsModeSupported;
+      m_instanceData->toAddon.stream_initialize = ADDON_StreamInitialize;
+
+      m_instanceData->toAddon.input_process = ADDON_InputProcess;
+
+      m_instanceData->toAddon.input_resample_process_needed_samplesize = ADDON_InputResampleProcessNeededSamplesize;
+      m_instanceData->toAddon.input_resample_process = ADDON_InputResampleProcess;
+      m_instanceData->toAddon.input_resample_get_delay = ADDON_InputResampleGetDelay;
+      m_instanceData->toAddon.input_resample_samplerate = ADDON_InputResampleSampleRate;
+
+      m_instanceData->toAddon.pre_process_needed_samplesize = ADDON_PreProcessNeededSamplesize;
+      m_instanceData->toAddon.pre_process_get_delay = ADDON_PreProcessGetDelay;
+      m_instanceData->toAddon.pre_process = ADDON_PreProcess;
+
+      m_instanceData->toAddon.master_process_set_mode = ADDON_MasterProcessSetMode;
+      m_instanceData->toAddon.master_process_needed_samplesize = ADDON_MasterProcessNeededSamplesize;
+      m_instanceData->toAddon.master_process_get_delay = ADDON_MasterProcessGetDelay;
+      m_instanceData->toAddon.master_process_get_out_channels = ADDON_MasterProcessGetOutChannels;
+      m_instanceData->toAddon.master_process = ADDON_MasterProcess;
+      m_instanceData->toAddon.master_process_get_stream_info_string = ADDON_MasterProcessGetStreamInfoString;
+
+      m_instanceData->toAddon.post_process_needed_samplesize = ADDON_PostProcessNeededSamplesize;
+      m_instanceData->toAddon.post_process_get_delay = ADDON_PostProcessGetDelay;
+      m_instanceData->toAddon.post_process = ADDON_PostProcess;
+
+      m_instanceData->toAddon.output_resample_process_needed_samplesize = ADDON_OutputResampleProcessNeededSamplesize;
+      m_instanceData->toAddon.output_resample_process = ADDON_OutputResampleProcess;
+      m_instanceData->toAddon.output_resample_get_delay = ADDON_OutputResampleGetDelay;
+      m_instanceData->toAddon.output_resample_samplerate = ADDON_OutputResampleSampleRate;
+    }
+
+    static inline void ADDON_GetCapabilities(AddonInstance_AudioDSP* instance, AE_DSP_ADDON_CAPABILITIES *capabilities)
```

Comment:
Right, but only on this place. The function direct on addon c++ class `virtual void GetCapabilities(AE_DSP_ADDON_CAPABILITIES& capabilities) = 0;` is with `&`